### PR TITLE
NF: Remove hook chess

### PR DESCRIPTION
**Pull request by Arthur-Milchior**
Restriction of #6268 restricted to Chess part for simplicity

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/libanki/hooks/Hooks.java [hunk 1]
```diff
@@ -22,6 +22,7 @@
 
 
 import com.ichi2.anki.AnkiDroidApp;
+import com.ichi2.libanki.Collection;
```
Comment (timrae) [inline AnkiDroid/src/main/java/com/ichi2/libanki/hooks/Hooks.java:25]: Codacy Issue found: [Avoid unused imports such as 'com.ichi2.libanki.Collection'](https://app.codacy.com/manual/timrae/Anki-Android/pullRequest?prid=5575768)

**Comment by timrae** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 5
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/timrae/Anki-Android/pullRequest?prid=5575768&bid=18280044)

File: AnkiDroid/src/main/java/com/ichi2/libanki/Collection.java [hunk 2]
```diff
@@ -2243,4 +2246,71 @@ public boolean getFailed() {
...
+                    "    fentxt=fentxt.replace(/P/g,'<td>&#9817;</td>');" +
+                    "    fentxt=fentxt.replace(/k/g,'<td>&#9818;</td>');" +
+                    "    fentxt=fentxt.replace(/q/g,'<td>&#9819;</td>');" +
+                    "    fentxt=fentxt.replace(/x/g,'<td>&#9820;</td>');" +
+                    "    fentxt=fentxt.replace(/b/g,'<td>&#9821;</td>');" +
+                    "    fentxt=fentxt.replace(/n/g,'<td>&#9822;</td>');" +
+                    "    fentxt=fentxt.replace(/p/g,'<td>&#9823;</td>');" +
+                    "    return '<div align=\"center\" width=\"100%%\"><table class=\"chess_board\" cellspacing=\"0\" cellpadding=\"0\"><tr>'+fentxt+'</tr></table></div>';" +
+                    "})('%s', %b)";
+    private static boolean sChessEnabled = false;
```
Comment (timrae) [inline AnkiDroid/src/main/java/com/ichi2/libanki/Collection.java:2282]: Codacy Issue found: [Fields should be declared at the top of the class, before any method declarations, constructors, initializers or inner classes.](https://app.codacy.com/manual/timrae/Anki-Android/pullRequest?prid=5575768)

File: AnkiDroid/src/main/java/com/ichi2/libanki/Collection.java [hunk 3]
```diff
@@ -2243,4 +2246,71 @@ public boolean getFailed() {
             return mFailed;
         }
     }
+
+    private static final Pattern fFenPattern = Pattern.compile("\\[fen ?([^\\]]*)\\]([^\\[]+)\\[/fen\\]");
```
Comment (timrae) [inline AnkiDroid/src/main/java/com/ichi2/libanki/Collection.java:2250]: Codacy Issue found: [Fields should be declared at the top of the class, before any method declarations, constructors, initializers or inner classes.](https://app.codacy.com/manual/timrae/Anki-Android/pullRequest?prid=5575768)

File: AnkiDroid/src/main/java/com/ichi2/libanki/Collection.java [hunk 4]
```diff
@@ -2243,4 +2246,71 @@ public boolean getFailed() {
...
+    private static final Pattern fFenOrientationPattern = Pattern.compile("orientation *= *\"?(black|white)\"?");
```
Comment (timrae) [inline AnkiDroid/src/main/java/com/ichi2/libanki/Collection.java:2251]: Codacy Issue found: [Fields should be declared at the top of the class, before any method declarations, constructors, initializers or inner classes.](https://app.codacy.com/manual/timrae/Anki-Android/pullRequest?prid=5575768)

File: AnkiDroid/src/main/java/com/ichi2/libanki/Collection.java [hunk 5]
```diff
@@ -2243,4 +2246,71 @@ public boolean getFailed() {
...
+    private static final String fRenderFen =
```
Comment (timrae) [inline AnkiDroid/src/main/java/com/ichi2/libanki/Collection.java:2252]: Codacy Issue found: [Fields should be declared at the top of the class, before any method declarations, constructors, initializers or inner classes.](https://app.codacy.com/manual/timrae/Anki-Android/pullRequest?prid=5575768)

File: AnkiDroid/src/main/java/com/ichi2/libanki/hooks/AdvancedStatistics.java [hunk 6]
```diff
@@ -99,7 +99,7 @@
 @SuppressWarnings({"PMD.AvoidThrowingRawExceptionTypes","PMD.ExcessiveClassLength","PMD.ExcessiveMethodLength",
                     "PMD.AvoidReassigningParameters","PMD.FieldDeclarationsShouldBeAtStartOfClass","PMD.SwitchStatementsShouldHaveDefault",
                     "PMD.NPathComplexity","PMD.OneDeclarationPerLine","PMD.SwitchStmtsShouldHaveDefault"})
-public class AdvancedStatistics extends Hook  {
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/libanki/hooks/AdvancedStatistics.java:102]: nit: unnecessary whitespace

File: AnkiDroid/src/main/java/com/ichi2/libanki/Collection.java (hunk 3 above)
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/libanki/Collection.java:2250]: From a design perspective, these would be better in `ChessFilter`. Is there a rationale for moving these to Collection?

Comment (Arthur-Milchior) [reply AnkiDroid/src/main/java/com/ichi2/libanki/Collection.java:2250]: dealing with chess is a part of _renderQA.
I should admit that I fail to see why rendering is in Collection. However, I believe that putting all of rendering together makes sens.

Comment (david-allison-1) [reply AnkiDroid/src/main/java/com/ichi2/libanki/Collection.java:2250]: I'd rather we moved it back - Single Responsibility Principle and all.

Our hands are tied with Anki Desktop compatibility in terms of splitting out `Collection` further, but we might as well make code intuitive for the next developers that come along

Comment (Arthur-Milchior) [reply AnkiDroid/src/main/java/com/ichi2/libanki/Collection.java:2250]: I don't know whether you check Anki's update or not. But you may be interested in knowing that _renderQA is not in collection.py. 
It's in rust back-end.
So by leaving _renderQA here, we are not following anki anymore

Comment (Arthur-Milchior) [reply AnkiDroid/src/main/java/com/ichi2/libanki/Collection.java]: I just removed the last commits

**Comment by timrae** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- AnkiDroid/src/main/java/com/ichi2/libanki/hooks/ChessFilter.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/timrae/Anki-Android/pullRequest?prid=5575768&bid=18280044)

File: AnkiDroid/src/main/java/com/ichi2/libanki/hooks/ChessFilter.java [hunk 7]
```diff
@@ -24,7 +24,7 @@
 
 import timber.log.Timber;
 
-public class ChessFilter extends Hook {
+public class ChessFilter {
```
Comment (timrae) [inline AnkiDroid/src/main/java/com/ichi2/libanki/hooks/ChessFilter.java:27]: Codacy Issue found: [The utility class name 'ChessFilter' doesn't match '[A-Z][a-zA-Z0-9]+(Utils?|Helper|Constants)'](https://app.codacy.com/manual/timrae/Anki-Android/pullRequest?prid=5575768)

File: AnkiDroid/src/main/java/com/ichi2/libanki/Storage.java [hunk 8]
```diff
@@ -47,6 +50,11 @@ public static Collection Collection(Context context, String path, boolean server
         assert path.endsWith(".anki2");
         // Since this is the entry point into libanki, initialize the hooks here.
         Hooks.getInstance(context);
+        SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(context.getApplicationContext());
+        // Preferences activated hooks
+        if (prefs.getBoolean("convertFenText", false)) {
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/libanki/Storage.java:55]: Also mentioned in the statistics PR, but I feel a static factory over static instance state would be a better option.

**Comment by Arthur-Milchior** [comment]
@david-allison-1 do you have any idea how to access a context here ? You solution worked in advanced stats because there was already a context. But here, I've got no context. The chess method is called by the card generation method, which has no context either. So I don't know how to correct this PR

**Comment by david-allison-1** [comment]
> @david-allison-1 do you have any idea how to access a context here ? You solution worked in advanced stats because there was already a context. But here, I've got no context. The chess method is called by the card generation method, which has no context either. So I don't know how to correct this PR

Collection has a local variable: `mContext`

**Comment by Arthur-Milchior** [comment]
Thanks.  I rebased following your idea.
It clearly lead to using less code
